routers/web/org/setting/actions.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package setting
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	actions_model "code.gitea.io/gitea/models/actions"
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	actions_shared "code.gitea.io/gitea/routers/web/shared/actions"
+)
+
+const (
+	// tplSettingsActions template path for render actions settings
+	tplSettingsActions base.TplName = "org/settings/actions"
+	// tplSettingsRunnersEdit template path for render runners edit settings
+	tplSettingsRunnersEdit base.TplName = "org/settings/runners_edit"
+)
+
+// Actions render settings/actions page for organization level
+func Actions(ctx *context.Context) {
+	pageType := ctx.Params(":type")
+	if pageType == "runners" {
+		ctx.Data["PageIsOrgSettingsRunners"] = true
+		page := ctx.FormInt("page")
+		if page <= 1 {
+			page = 1
+		}
+
+		opts := actions_model.FindRunnerOptions{
+			ListOptions: db.ListOptions{
+				Page:     page,
+				PageSize: 100,
+			},
+			Sort:          ctx.Req.URL.Query().Get("sort"),
+			Filter:        ctx.Req.URL.Query().Get("q"),
+			OwnerID:       ctx.Org.Organization.ID,
+			WithAvailable: true,
+		}
+
+		actions_shared.RunnersList(ctx, opts)
+	} else if pageType == "secrets" {
```

Comment:
I do not understand why the "runner" and "secrets" should share the same handler.

Reply:
Because they are sub settings below actions, and both of them are not visible if `actions.MustEnableActions` is false.

Reply:
I do not see why they couldn't have separate handlers. Separate handlers could also be protected by one `actions.MustEnableActions`

Reply:
I see what you mean, I will separate them

Reply:
Added a commit https://github.com/go-gitea/gitea/pull/24200/commits/2b086fd76182b79e87af646a1fe99ace57266bc2 to seoarate them